домашнее задание hw08 Обработчик json-ов

DmitryLapitsky opened the pull request:
Домашнее задание 08: Обработчик json-ов
Почему-то не захотело определяться в тестах readString
Files.readString(Paths.get(fullOutputFilePath));
Поменял на 
new String(Files.readAllBytes(path))

DmitryLapitsky committed:
домашнее задание hw08 Обработчик json-ов

petrelevich commented on hw08_json/src/test/java/ru/otus/dataprocessor/ProcessorTest.java:
```diff
@@ -0,0 +1,53 @@
+        System.out.println("tempDir\t" + tempDir);
+
+        //given
+        var inputDataFileName = "D:\\SelfLearn\\OTUS\\2020_12_OTUS_java_Lapitsky\\hw08_json\\src\\test\\resources\\inputData.json";// "inputData.json";
```
обратите внимание.
файл с данными лежал в ресурсах и работать с ним надо как с файлом в ресурсах.

DmitryLapitsky replied:
Здравствуйте, Сергей. Простите, в попыхах не заметил эту ошибку. Переделал в Paths.get("src", "test","resources", "inputData.json")

petrelevich commented on hw08_json/src/main/java/ru/otus/dataprocessor/ProcessorAggregator.java:
```diff
@@ -0,0 +1,26 @@
+
+
+        for (Measurement m : data) {
+            if (!map.keySet().contains(m.getName())) {
```
попробуйте это переписать с использованием stream api.

DmitryLapitsky replied:
Переделал через stream api. Правда внутри все равно пришлось if добавлять, только используя методы api не вышло:(

DmitryLapitsky committed:
Поправил в тестах работу с расположением тестового файла - переделал в Paths.get()
При заполнении map в классе ProcessorAggregator использовал stream api

DmitryLapitsky committed:
убрана лишняя печать в консоль

petrelevich commented on hw08_json/src/test/java/ru/otus/dataprocessor/ProcessorTest.java:
```diff
@@ -0,0 +1,48 @@
+    void processingTest(@TempDir Path tempDir) throws IOException {
+
+        //given
+        var inputDataFileName = Paths.get("src", "test","resources", "inputData.json").toFile().getPath();
```
В оригинале было так:
` var inputDataFileName = "inputData.json";`
Давайте с этим вариантом сделаем.

прочитать такой файл можно так:
`var resource = getClass().getClassLoader().getResource(fileName);`

DmitryLapitsky replied:
Переделал в 
var inputDataFileName = "inputData.json";
URL resource = getClass().getClassLoader().getResource(inputDataFileName);
FileLoader loader = new FileLoader(new File(Objects.requireNonNull(resource).toURI()).getAbsolutePath());

petrelevich commented on hw08_json/src/main/java/ru/otus/dataprocessor/ProcessorAggregator.java:
```diff
@@ -0,0 +1,24 @@
+                map.put(el.getName(), map.get(el.getName()) + el.getValue());
+            }
+        });
+        return map;
```
Можно ли использовать такую заготовку ?
```
        data.stream()
                .collect(Collectors.groupingBy(Measurement::getName, Collectors.summingDouble(Measurement::getValue)));
```

DmitryLapitsky replied:
Спасибо. Я бы долго еще искал, как это сделать. Реализовал с помощью 
new TreeMap<>(data.stream()
                .collect(Collectors.groupingBy(
                        Measurement::getName,
                        Collectors.summingDouble(Measurement::getValue))
                )
        );

DmitryLapitsky committed:
Работа над ошибками:
в ProcessorAggregator map собирается только используя stream api
в тесте ProcessorTest определяется текущая деректория и файл считывается через URI
